### PR TITLE
Fix Flycap & Spinnaker endpoints

### PR DESCRIPTION
**Pull request by civerachb-cpr**

### CODE REVIEW

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (mikepurvis) [inline spinnaker_camera_driver/cmake/download_spinnaker:44]: If we're now in the clear to host this, all of the downloading shenanigans can be dropped in favour of asking OSRF to just host the debs on their mirror. Make a PR here:

https://github.com/ros-infrastructure/reprepro-updater/tree/master/config

Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker:44]: The packages are not currently hosted on an actual deb server.  There's some question about whether or not it's worth setting one up to host them vs the immediate this-works-the-same-as-it-always-has downloading shenanigans.
Also, for the PointGrey camera driver (https://github.com/ros-drivers/pointgrey_camera_driver), the armhf and arm64 archives do not contain deb packages at all, so what you're suggesting isn't an option for that combination of package & architectures.

Comment (mikepurvis) [reply spinnaker_camera_driver/cmake/download_spinnaker:44]: Fair. If we don't want to go to the trouble of packaging the ARM assets for OSRF to host, another option would be to do a one-time unpack of this stuff and commit the headers and binary blobs into this repo (or script it so that that step can be easily rerun later on when Spinnaker gets updated).

I just know that in general, accessing external network resources during build is considered an anti-pattern, and this specific instance of it has caused issues in the past both with the public buildfarm and OTTO's bundle builder.

**Comment by ghost** [comment]
I agree with @mikepurvis but I'm ok with merging this in favor of having a working version available. @civerachb-cpr before we merge can you run a prerelease test on your machine?

**Comment by mikepurvis** [comment]
Yeah, I don't really have a dog in this fight; it's up to you guys what you want to do here. But it would be nice to see this cleaned up, if not now, then at least at some point.

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (ghost) [inline spinnaker_camera_driver/cmake/download_spinnaker:55]: Should we remove this field? it doesn't seem to be used.

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (ghost) [inline spinnaker_camera_driver/cmake/download_spinnaker]: This deb doesn't seem to be included any more.  If the package works without it this line should be removed. It may have become libspinvideo.

**Comment by ghost** [comment]
Also it may be worth stripping down the tars that we host. ~90MB is a bunch to download. The build farm may time out downloading it.  We could remove doc's at the very least.

**Comment by civerachb-cpr** [comment]
I've been asked by other people on my end if we can just move ahead with this fix, as the FLIR drivers currently don't built at all, and this patch resolves the immediate problem.  We can debate the merits of building ARM* .deb packages & hosting everything on a PPA in another ticket.

I'll see what I can do about stripping out the unnecessary data from the archives for now to prevent any download issues.  I don't have permission to upload anything to that particular server, so I can't give you a firm timeline on when that'll actually get approved though.

EDIT:
As far as I'm aware, we were always downloading the whole archive of packages previously though, weren't we?  The majority of the changes I made are to the endpoint itself (new location, removal of the authentication/login portion).  So if the build farm worked before with a 90MB download it should continue to work now, shouldn't it?

**Comment by mikepurvis** [comment]
I don't have a strong opinion on any of this. I'm fine with it for now. If you haven't got permissions, we should make one or both of you guys a maintainer, as we're not using this package at present.

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (ghost) [inline spinnaker_camera_driver/cmake/download_spinnaker:185]: Need to create this folder if it doesn't exist.

File: spinnaker_camera_driver/cmake/DownloadSpinnaker.cmake
Comment (ghost) [inline spinnaker_camera_driver/cmake/DownloadSpinnaker.cmake:13]: why wouldn't we use the library directory?

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (ghost) [inline spinnaker_camera_driver/cmake/download_spinnaker:199]: Do we only need one of the libraries? Do we not need the xml files?

**Review by ghost — CHANGES_REQUESTED** [review]
Added some notes after running a pre-release test

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker:199]: The original version didn't include the XML files either, so I assume they are not necessary.

File: spinnaker_camera_driver/cmake/DownloadSpinnaker.cmake
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/DownloadSpinnaker.cmake:13]: Because this is the folder that the original version of the script used.
These modifications were made from the perspective of "the endpoint where we download the packages from is different; fix that, but don't modify the end behaviour of anything else unless it's strictly necessary."

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker]: When I build the packages with catkin_make I don't get any errors related to the directory not existing.  I can add an explicit check & make the directories manually, but it feels like its not necessary?

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker]: Removed.

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker:55]: This was introduced to keep the formatting consistent between the flir + spinnaker drivers & the pointgrey/flycap driver, which has the same build problems & was fixed at the same time.
The flycap driver has some code delivered as .debs and others as pre-compiled but un-packaged archives.
Having extra fields defined here doesn't actively hamper anything and it keeps the formatting consisted across projects.

File: spinnaker_camera_driver/cmake/download_spinnaker
Comment (civerachb-cpr) [reply spinnaker_camera_driver/cmake/download_spinnaker:185]: Fixed.

**Comment by stwirth** [comment]
:+1: for getting this merged (and released) soon.

**Pull request merged**
